### PR TITLE
Remove dependency on lsb_release

### PR DESCRIPTION
**Pull request by ktdreyer**
The `lsb_release` utility brings in a lot of other dependencies. Remove it from the tree.

Fixes: https://tracker.ceph.com/issues/52243

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/workunits/rgw/s3_utilities.pm [hunk 1]
```diff
@@ -134,15 +134,31 @@ sub purge_data
...
+{
+        open(FH, '<', '/etc/os-release') or die $!;
+        while (my $line = <FH>) {
+                if($line =~ /^PRETTY_NAME=(.*)$/)
+                {
+                        my $pretty_name = $1;
+                        $pretty_name =~ s/^"//;
+                        $pretty_name =~ s/"$//;
+                        return $pretty_name;
+                }
```
Comment (tchaikov) [inline qa/workunits/rgw/s3_utilities.pm:148]:
```suggestion
                if ($line =~ /^\s*PRETTY_NAME=\"?(.*)\"/) {
                    return $1;
                }
```

File: qa/workunits/rgw/s3_utilities.pm [hunk 2]
```diff
@@ -134,15 +134,31 @@ sub purge_data
     return 0;
 }
 
+# Read PRETTY_NAME from /etc/os-release
+sub os_pretty_name
+{
+        open(FH, '<', '/etc/os-release') or die $!;
```
Comment (tchaikov) [inline qa/workunits/rgw/s3_utilities.pm:140]: might want to use 4 space for indent to be more consistent with the rest of this source file.

**Comment by idryomov** [comment]
For reference:

```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

```
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="8 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Linux 8 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-8"
CENTOS_MANTISBT_PROJECT_VERSION="8"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="8"
```

```
$ cat /etc/os-release
NAME="CentOS Stream"
VERSION="8"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Stream 8"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_SUPPORT_PRODUCT_VERSION="CentOS Stream"
```

**Comment by tchaikov** [comment]
https://pulpito.ceph.com/kchai-2021-08-13_12:30:46-rados-wip-kefu-testing-2021-08-13-1658-distro-basic-smithi/

cephadm failures are tracked by https://tracker.ceph.com/issues/52279

**Comment by idryomov** [comment]
rbd workunit change looks good!

https://pulpito.ceph.com/dis-2021-08-16_11:32:48-rbd-wip-dis-testing-distro-basic-smithi/
https://pulpito.ceph.com/dis-2021-08-16_13:18:46-rbd-wip-dis-testing-distro-basic-smithi/

**Review by idryomov — APPROVED** [review]

File: qa/workunits/rgw/s3_utilities.pm (hunk 1 above)
Comment (ktdreyer) [reply qa/workunits/rgw/s3_utilities.pm:148]: with this change, the method returns `Fedora 34 (Thirty Four)"` on my system... still investigating...

Comment (ktdreyer) [reply qa/workunits/rgw/s3_utilities.pm:148]: Oh, yeah that `.*` glob is greedy, so it needs to exclude `"` characters.

**Review by tchaikov — APPROVED** [review]

**Comment by ideepika** [comment]
https://pulpito.ceph.com/?branch=wip-deepika2-testing-2021-08-27-0331
https://tracker.ceph.com/issues/52398
https://tracker.ceph.com/issues/49287
https://tracker.ceph.com/issues/52315
https://tracker.ceph.com/issues/52447

**Comment by idryomov** [comment]
Backported to octopus in https://github.com/ceph/ceph/pull/45206.

**Pull request merged**